src/RA_ImageFactory.cpp

```diff
@@ -54,32 +53,35 @@ HRESULT UserImageFactory_CreateDIBSectionFromBitmapSource(IWICBitmapSource *pToR
     // Note that the height is negative for top-down bitmaps
     if (SUCCEEDED(hr))
     {
-        ZeroMemory(&bminfo, sizeof(bminfo));
-        bminfo.bmiHeader.biSize = sizeof(BITMAPINFOHEADER);
-        bminfo.bmiHeader.biWidth = nWidth;
-        bminfo.bmiHeader.biHeight = -(LONG)nHeight;
-        bminfo.bmiHeader.biPlanes = 1;
-        bminfo.bmiHeader.biBitCount = 32;
-        bminfo.bmiHeader.biCompression = BI_RGB;
-
-        hWindow = GetActiveWindow();
+
+
+		// TODO: Maybe replace some of these with literals
+		BITMAPINFOHEADER info_header{
+			sizeof(BITMAPINFOHEADER),    // biSize
+            static_cast<LONG>(nWidth),
+			-static_cast<LONG>(nHeight),
+            WORD{1},                     // biPlanes
+            WORD{32},                    // biBitCount
+			static_cast<DWORD>(BI_RGB)
+		};
+		BITMAPINFO bminfo{ info_header };
+
+		auto hWindow = GetActiveWindow();
         while (GetParent(hWindow) != nullptr)
             hWindow = GetParent(hWindow);
 
         // Get a DC for the full screen
-        hdcScreen = GetDC(hWindow);
-        hr = hdcScreen ? S_OK : E_FAIL;
+		auto hdcScreen = GetDC(hWindow);
+
 
         // Release the previously allocated bitmap 
-        if (SUCCEEDED(hr))
+		if (hr = hdcScreen ? S_OK : E_FAIL; SUCCEEDED(hr))
```

Comment:
I realize this is how it was before, but suggest simplifying it to:
```
if (hdcScreen)
```
there's no reason to get `hr` involved.

Reply:
Thanks for the suggestion.